plumbing/transport/ssh/common.go

```diff
@@ -129,3 +140,25 @@ func (c *command) setAuthFromEndpoint() error {
 func endpointToCommand(cmd string, ep transport.Endpoint) string {
 	return fmt.Sprintf("%s '%s'", cmd, ep.Path())
 }
+
+func overrideConfig(overrides *ssh.ClientConfig, c *ssh.ClientConfig) {
+	if overrides == nil {
+		return
+	}
+
+	vo := reflect.ValueOf(*overrides)
```

Comment:
But we don't want to override values set for the session, such as `HostKeyCallback`.